src/shared/telemetry/defaultTelemetryClient.ts

```diff
@@ -17,14 +17,31 @@ import { ServiceConfigurationOptions } from 'aws-sdk/lib/service'
 import { DefaultSettingsConfiguration } from '../settingsConfiguration'
 import globals from '../extensionGlobals'
 
-export class DefaultTelemetryClient implements TelemetryClient {
-    public static readonly DEFAULT_IDENTITY_POOL = 'us-east-1:820fd6d1-95c0-4ca4-bffb-3f01d32da842'
-    public static readonly DEFAULT_TELEMETRY_ENDPOINT = 'https://client-telemetry.us-east-1.amazonaws.com'
+interface TelemetryConfiguration {
+    readonly endpoint: string
+    readonly identityPool: string
+}
 
+export class DefaultTelemetryClient implements TelemetryClient {
+    private static readonly DEFAULT_IDENTITY_POOL = 'us-east-1:820fd6d1-95c0-4ca4-bffb-3f01d32da842'
+    private static readonly DEFAULT_TELEMETRY_ENDPOINT = 'https://client-telemetry.us-east-1.amazonaws.com'
     private static readonly PRODUCT_NAME = 'AWS Toolkit For VS Code'
 
+    private static initializeConfig(): TelemetryConfiguration {
+        const settings = new DefaultSettingsConfiguration()
+        const identityPool = settings.readDevSetting<string>('aws.dev.telemetryUserPool', 'string', true)
+        const endpoint = settings.readDevSetting<string>('aws.dev.telemetryEndpoint', 'string', true)
```

Comment:
Is there a reason for this? I'm kinda thinking that if someone wanted to set up their own telemetry endpoint/user pool, it'd be something they'd want settable outside the context of a dev setting.

Reply:
Oh I just added it since I was working on my own telemetry stack. We already set endpoints for other APIs so it does fit, though honestly it'd be one of the least used settings.